src/EPICS-Backend.cc

```diff
@@ -271,13 +267,16 @@ namespace ChimeraTK {
     ChannelManager::getInstance().waitForConnections(_caTimeout);
   }
 
-  void EpicsBackend::setException() {
-    //\ToDo: Why I have to check is functional here? If not setException is called constantly and which means
-    //_isFunctional will stay false even if reset in the state handler.
-    if(_isFunctional) {
-      _isFunctional = false;
-      _asyncReadActivated = false;
-      ChannelManager::getInstance().setException(std::string("Exception reported by another accessor."));
-    }
+  void EpicsBackend::setExceptionImpl() noexcept {
+    /// FIXME: This is not thread safe for several reasons!
+    /// 1. _asyncReadActivated is a plain bool variable, so some thread synchronisation mechamism is missing (e.g.
+    ///    convert into std::atomic).
+    /// 2. We need the guarantee that nothing writes to the accessor queues beyond this point. In the
+    ///    EpicsBackendRegisterAccessorBase, the _asyncReadActivated flag is first checked and then the queue is filled.
+    ///    There is no guarantee that in between the check and the queue filling this function is called. This would
+    ///    violate the specification and confuse e.g. ApplicationCore exception handling.
+    _asyncReadActivated = false;
+    /// FIXME: There is missing code here! The queues of all accessors need to be filled with exceptions (if asyncRead
```

Comment:
The previous version was calling ChannelManager::getInstance().setException(), and only if the device was functional , which probably does exactly what is needed (except for the thread safety). This is missing in the new code.

Reply:
So far I did not push the branch I'm currently working on (I will do now). It includes the UnifiedBackendTest and should comply to the backend specifications. Please fix the backend on that branch and I will merge it after...

Reply:
Branch name is: https://github.com/ChimeraTK/DeviceAccess-EpicsBackend/tree/add_test